raiden-dapp/src/App.vue

```diff
@@ -22,19 +22,42 @@
 </template>
 
 <script lang="ts">
-import { Component, Vue } from 'vue-property-decorator';
+import { Component, Watch, Mixins } from 'vue-property-decorator';
+import { mapState } from 'vuex';
+import { DeniedReason } from '@/model/types';
+import NavigationMixin from './mixins/navigation-mixin';
 import AppHeader from '@/components/AppHeader.vue';
 import OfflineSnackbar from '@/components/OfflineSnackbar.vue';
 import UpdateSnackbar from '@/components/UpdateSnackbar.vue';
 
 @Component({
+  computed: mapState(['loading', 'accessDenied']),
   components: {
     AppHeader,
     OfflineSnackbar,
     UpdateSnackbar
   }
 })
-export default class App extends Vue {
+export default class App extends Mixins(NavigationMixin) {
+  loading!: boolean;
+  accessDenied!: DeniedReason;
+
+  @Watch('loading', { immediate: true })
+  onLoadingChange() {
+    this.redirectOnReload();
+  }
+
+  @Watch('accessDenied', { immediate: true })
+  onAccessDeniedChange() {
+    this.redirectOnReload();
+  }
+
+  redirectOnReload() {
+    if (this.accessDenied !== DeniedReason.UNDEFINED || this.loading) {
+      this.navigateToHome();
+    }
+  }
+
   destroyed() {
     this.$raiden.disconnect();
```

Comment:
Navigating to home after this disconnect didn't work probably? I'm wondering how it worked earlier